compiler/tests-compiler/gh1659.ml

```diff
@@ -0,0 +1,9 @@
+let%expect_test _ =
+  let prog = {|
+let x = (0., 0.) = (-0., 0.);;
```

Comment:
I would also add a test for `Js.equals (0, 0) (0, 0)`.

Reply:
You can explicit add an external inside `prog`

```
external equals : 'a -> 'a -> bool = "caml_js_equals"
let x = equals (0, 0) (0, 0)
Printf.printf "%B\n" x;;
```

Reply:
Done